services/frontend-service/src/ui/utils/store.tsx

```diff
@@ -487,7 +487,9 @@ export const useCurrentlyDeployedAtGroup = (application: string, version: number
 // Get release information for a version
 export const useReleaseInfo = (app: string, version: number): Release =>
     useOverview(({ applications }) => {
-        const releaseInfo = applications[app]?.releases.filter((release) => release.version === version)[0];
+        const releaseInfo = applications[app]?.releases.filter((r) =>
+            version === -1 ? r.undeployVersion : r.version === version
+        )[0];
         if (!releaseInfo) {
             return {} as Release;
```

Comment:
please don't use `as`.
If there's nothing found we should return undefined.

Reply:
:+1:

Reply:
that is why I added functions like `useReleaseOptional`

Reply:
I will leave that for now. and will comeback to it later when there is more time.